modules/platform/login/nuxeo-platform-login-token-rest/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/login/tokenauth/AuthenticationTokensObject.java

```diff
@@ -65,12 +65,33 @@ protected void initialize(Object... args) {
     }
 
     @POST
+    @Produces(MediaType.TEXT_PLAIN)
     public Response createToken(@QueryParam("application") String applicationName,
             @QueryParam("deviceId") String deviceId, @QueryParam("deviceDescription") String deviceDescription,
             @QueryParam("permission") String permission) {
         String username = getCurrentUser().getName();
         String token = service.acquireToken(username, applicationName, deviceId, deviceDescription, permission);
-        return Response.ok(token).status(Response.Status.CREATED).build();
+        return Response.ok(token).type(MediaType.TEXT_PLAIN_TYPE).status(Response.Status.CREATED).build();
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createJSONToken(@QueryParam("application") String applicationName,
+            @QueryParam("deviceId") String deviceId, @QueryParam("deviceDescription") String deviceDescription,
+            @QueryParam("permission") String permission, @QueryParam("detail") boolean detail) {
+        String username = getCurrentUser().getName();
+        String token = service.acquireToken(username, applicationName, deviceId, deviceDescription, permission);
+        Object body = String.format("\"%s\"", token);
+        if (detail) {
+            // Get the token from the service to populate additional fields like 'creationDate'
+            body = getTokens(applicationName).stream()
+                                             .filter(item -> token.equals(item.getToken()))
+                                             .findFirst()
+                                             // Erase AuthenticationToken type signature
```

Comment:
The comment is a bit distracting when reading the code, especially since it describes what the code does and not why. I'd simply remove it, people interested in understanding why can see in the IDE the error reported if they remove the cast.

Reply:
Removed the comment.